setup.py

```diff
@@ -124,17 +124,26 @@ def get_output(*args, **kwargs):
 
 # add in the extra workarounds for different include structures
 if winkrb_path:
-    prefix = winkrb_path
+    prefixes = [winkrb_path]
 else:
+    prefixes = []
     try:
-        prefix = get_output(f"{kc} gssapi --prefix")
+        prefixes.append(get_output(f"{kc} gssapi --prefix"))
     except Exception:
-        print("WARNING: couldn't find krb5-config; assuming prefix of %s"
-              % str(sys.prefix))
-        prefix = sys.prefix
-gssapi_ext_h = os.path.join(prefix, 'include/gssapi/gssapi_ext.h')
-if os.path.exists(gssapi_ext_h):
-    compile_args.append("-DHAS_GSSAPI_EXT_H")
+        print("WARNING: couldn't find krb5-config to determine prefix")
+    try:
+        prefixes.append(os.path.dirname(
+            get_output("pkg-config --variable=includedir krb5-gssapi"))
+        )
+    except Exception:
+        print("WARNING: could not find pkg-config to determine includedir")
+    prefixes.append(sys.prefix)
+
+for prefix in prefixes:
+    gssapi_ext_h = os.path.join(prefix, 'include/gssapi/gssapi_ext.h')
```

Comment:
On Fedora 38 the `pkg-config` output is `/usr/include` so the check would ultimately be for `/usr/include/include/gssapi/gssapi_ext.h`. It will still work because as you said it will check the `krb5-config gssapi --prefix` first but it sounds like the logic here might need some extra work.

Even testing on SLES 15 shows that it returns `/usr/include` so this fix won't work on the platforms that might benefit from it a bit more.

Reply:
The call to `os.path.dirname` is what was wrong, will push a fixed version in a few minutes.